### PR TITLE
Improve CustomActiveRecordObserver to work with Paranoia and Awesome Nested Set gems

### PR DESCRIPTION
**Pull request by lesykgg**

### CODE REVIEW

**Review by dkh — APPROVED** [review]

**Review by myacheg — APPROVED** [review]
Nice one! Good job!

File: test.sh [hunk 1]
```diff
@@ -3,6 +3,10 @@
 echo "*** Running custom_active_record_observer specs"
 
 bundle install                                      || exit 1
+bundle exec appraisal install                       || exit 1
```
Comment (vbyno) [inline test.sh]: Maybe, remove `bundle install` and `bundle exec appraisal install` from here and add them to `cache` options in `travis.yml`?

File: test.sh [hunk 2]
```diff
@@ -3,6 +3,10 @@
...
 bundle exec rake app:db:drop app:db:create          || exit 1
 bundle exec rake app:db:migrate app:db:test:prepare || exit 1
-bundle exec rspec spec
+
+bundle exec rspec spec --exclude-pattern 'spec/apprisals/**/*_spec.rb' &&
```
Comment (vbyno) [inline test.sh]: doesn't `--exclude-pattern 'spec/apprisals/**'` work?

Comment (lesykgg) [reply test.sh]: it doesn't

**Review by vbyno — APPROVED** [review]

**Pull request merged**